gateway/src/apicast/policy/rewrite_url_captures/rewrite_url_captures.lua

```diff
@@ -0,0 +1,73 @@
+--- Rewrite URL Captures policy
+-- This policy captures arguments in a URL and rewrites the URL using those
+-- arguments.
+-- For example, we can specify a matching rule with arguments like
+-- '/{orderId}/{accountId}' and a template that specifies how to rewrite the
+-- URL using those arguments, for example:
+-- '/sales/v2/{orderId}?account={accountId}'.
+-- In that case, the request '/123/456' will be transformed into
+-- '/sales/v2/123?account=456'
+
+local NamedArgsMatcher = require('named_args_matcher')
+local QueryParams = require('apicast.query_params')
+
+local ipairs = ipairs
+local insert = table.insert
+
+local policy = require('apicast.policy')
+local _M = policy.new('Capture args policy')
+
+local new = _M.new
+
+function _M.new(config)
+  local self = new(config)
+
+  self.matchers = {}
```

Comment:
Just for the future. Using `tab_new(#config.transformations)` would save table expansion because it would preallocate slots in the table for each matcher.